enhancements/builds/node-cache-for-builds.md

```diff
@@ -0,0 +1,164 @@
+---
+title: node-cache-for-builds
+authors:
+  - "@adambkaplan"
+reviewers:
+  - "@nalind
+  - "@rhdan"
+  - "@TomSweeneyRedHat"
+  - "@bparees"
+approvers:
+  - "@bparees"
+creation-date: 2019-02-18
+last-updated: 2019-02-18
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Node Cache for Builds
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+  [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+None.
+
+## Summary
+
+Builds in OpenShift 4.x currently require every layer of the base image to be pulled from its source
+registry every time the build is run. In OCP 3.x builds could take advantage of image layers that
+were stored on the build's node. This proposal provides a means of partially restoring this
+node-level cache by allowing build pods to avoid pulling layers that are present on the node. Unlike
+3.x, however, layers created during the build process will not be directly cached on the node due to
+buildah's isolation.
+
+## Motivation
+
+### Goals
+
+1. Provide a means for builds to take advantage of image layers that are stored on the node.
+
+### Non-Goals
+
+1. Provide a means for builds to write their layers to a cache (node or shared).
+2. Provide a means for builds to share an image layer cache within a namespace.
+3. Provide a means for the internal registry to pre-populate an image layer cache within a namespace
+   or across a cluster.
+4. Add persistent volume support for builds.
+
+## Proposal
+
+Build pods will mount the image layer cache that exists on every OpenShift node into the build pod.
+Buildah - running within the build pod - will use this layer cache to reduce the number of layers
+that need to be pulled when the build commences.
+
+### User Story
+
+As a developer using OpenShift to build container images I want builds to use image layers cached on
+the node So that builds don't have to pull all layers in their base images And builds run faster.
+
+### Implementation Details/Notes/Constraints [optional]
+
+1. The build controller will mount the node image cache located at `/var/lib/containers/storage`
+   into the build pod as `readOnly`. The mount point should not conflict with buildah's own default
+   storage directory. Ex: `/var/lib/shared/containers/storage`.
```

Comment:
@adambkaplan - am I correct in thinking that if  in tekton/buildv2, where the `buildah` cli is directly invoked in one tekton `corev.Container` 's (the tekton `Step`) that use the `buildah` image, then the user simply needs to set the appropriate `buildah bud` option (perhaps `--cache-from` ? ... or is that a question for @nalind or @TomSweeneyRedHat  ?)

Anyway, understanding the CLI consumption mechanism in addition to the API ones done in openshift/builder is an important end goal here, IMO, as we discussed in planning as part of facilitating tekton/buildv2

Reply:
From @rhatdan's blog post [1], this isn't a CLI option, but a config option within `storage.conf`.

[1] https://developers.redhat.com/blog/2019/08/14/best-practices-for-running-buildah-in-a-container/